tests/adapters/test_httpadapter.py

```diff
@@ -91,14 +91,25 @@ async def adapter_task(
     mock_device: Device,
     event_loop: asyncio.BaseEventLoop,
 ):
-    task = event_loop.create_task(
+    adapter_running = event_loop.create_task(
         adapter.run_forever(mock_device, mock_raise_interrupt)
     )
-    await adapter.wait_until_ready()
-    yield task
+    adapter_ready = event_loop.create_task(adapter.wait_until_ready())
+
+    # either wait until the task has an exception or it's ready.
+    done, _ = await asyncio.wait(
+        [adapter_running, adapter_ready], return_when=asyncio.tasks.FIRST_COMPLETED
+    )
+
+    if adapter_running in done:
+        exception = adapter_running.exception()
+        if exception is not None:
```

Comment:
The adapter_running task should never be done. If it is, it's because an exception has been raised. So really, there may be a cleaner way of specifying that in such a situation, exception is always going to be not None. This is because adapter_running should run forever; if it isn't running forever (and has finished) an exception should be propagated.

Raising an exception here is a good idea, in case of future developments that change how run_forever works (which sounds likely) such that it can be stopped, for example.